[AIDIR] Internal util function to insert MLHub debugging insight for dynamic shape

mengtian-x opened the pull request:
Summary:
This feature is Meta internal only
Add a util function to put dynamic shape-related suggestion to MLHubDebugInsightService, which will then be surfaced to users in the MLHub .

The rollout will be controlled by JK.

Test Plan:


MAST job aps-omnifmv3_dev_baseline_test-a34fdccf21 

 {F1980593060}

* If you're not able to see the insight, please add yourself to this gk 'mlhub_debugging_insights_dev_visibility'
* The URL link should route to a new Job Inspector page that will provide details and straight forward instructions of how to config the ds. The page is currently still in development so here we use the general PT2 compile JI page. 
* Test fails because of the export checks. I'll export after addressing all the comments from reviewers.

Rollback Plan:

Reviewed By: pianpwk

Differential Revision: D78526522




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

jingsh reviewed (approved):
nit, can you fix the error

mengtian-x commented on torch/_utils_internal.py:
```diff
@@ -116,6 +116,8 @@ def wrapper_function(*args: _P.args, **kwargs: _P.kwargs) -> _T:
 def signpost_event(category: str, name: str, parameters: dict[str, Any]):
     log.info("%s %s: %r", category, name, parameters)
 
+def add_mlhub_insight(category: str, insights: str, insights_description: str):
```
Thanks for pointing out! Yes, I missed here when fixing a typo and it is addressed in the next revision.

mengtian-x committed:
[AIDIR] Internal util function to insert MLHub debugging insight for dynamic shape (#159391)

Summary:
This feature is Meta internal only
Add a util function to put dynamic shape-related suggestion to MLHubDebugInsightService, which will then be surfaced to users in the MLHub .

The rollout will be controlled by JK.


Test Plan:
MAST job aps-omnifmv3_dev_baseline_test-a34fdccf21 

 {F1980593060}

* If you're not able to see the insight, please add yourself to this gk 'mlhub_debugging_insights_dev_visibility'
* The URL link should route to a new Job Inspector page that will provide details and straight forward instructions of how to config the ds. The page is currently still in development so here we use the general PT2 compile JI page. 
* Test fails because of the export checks. I'll export after addressing all the comments from reviewers.

Rollback Plan:

Reviewed By: pianpwk

Differential Revision: D78526522

Pulled By: mengtian-x

mengtian-x commented:
@pytorchbot  merge
